File: packages/var/src/parse.ts
import type { Bdd } from './ast.js'
import { type ScannerPlugin, scan } from './scanner.js'
import { structure } from './structurer.js'

export function parse(
  path: string,
  source: string,
  plugins: ReadonlyArray<ScannerPlugin> = [],
): Bdd {
  return structure(path, source, scan(source, plugins))
}

File: python/packages/var-core/src/var_core/parse.py
"""parse.py — port of typescript/packages/var-core/src/parse.ts.

Combines scan + structure into the top-level parse function.
"""

from __future__ import annotations

from var_core.ast import VarDoc
from var_core.scanner import ScannerPlugin, scan
from var_core.structurer import structure


def parse(
    path: str,
    source: str,
    plugins: tuple[ScannerPlugin, ...] = (),
) -> VarDoc:
    """Parse *source* into a VarDoc: scan blocks then group them into Examples."""
    return structure(path, source, scan(source, plugins))
